.github/workflows/ingestion_client.yaml

```diff
@@ -31,8 +34,55 @@ jobs:
             echo "Generated ARM template does not match the existing main.json. Please update main.json accordingly." >&2
             exit 1
           fi
-  build:
+
+  test:
+    runs-on: ubuntu-latest
     needs: convert_bicep
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: '8.0.x'
+
+      - name: Install coverlet
+        run: dotnet tool install --global coverlet.console
+
+      - name: Install reportgenerator
+        run: dotnet tool install --global dotnet-reportgenerator-globaltool
+
+      - name: Run Unit Tests and Collect Coverage
+        run: |
+          dotnet test ./samples/ingestion/ingestion-client/Tests --configuration Release /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura
```

Comment:
I think we need to filter based on TestCategory here to skip e2e tests

Reply:
A previous change has mocked out the dependency in the E2E Test and it now runs locally? Maybe the change is to remove that TestCategory? Thoughts